ddtrace/ddtrace.go

```diff
@@ -50,6 +51,9 @@ type Tracer interface {
 
 	// Stop stops the tracer. Calls to Stop should be idempotent.
 	Stop()
+
+	// DataStreamsProcessor returns the data streams processor
+	DataStreamsProcessor() dsminterface.Processor
```

Comment:
Unfortunately, we cannot add methods to our public interfaces without breaking version compatibility. We will have to find a work-around for this. Possibly with "optional" interfaces.

i.e.
```
// preferably lives in the dsminterface package
type DataStreamProcessor interface {
    DataStreamsProcessor() dsminterface.Processor
}

// assume t is a ddtrace.Tracer
if dsp, ok := t.(DataStreamProcessor); ok {
    dsp.DataStreamsProcessor()
}
```

Alternatively, since this looks like it's only used in `ddtrace/tracer`, we could just type-assert to `*tracer`.

Happy to hear other options.

Reply:
Type-asserting to (*tracer) makes sense to me